start.sh

```diff
@@ -2,59 +2,107 @@
+    sed -i "s,  # email.username = vivtroAdmin@mydomain.edu,email.username = $EMAIL_USERNAME,g" /tmp/runtime.properties
+  fi
+  if [[ ! -z "${EMAIL_PASSWORD}" ]]; then
+    echo "Templating runtime.properties email.password = ***"
+    sed -i "s,  # email.password = secret,email.password = $EMAIL_PASSWORD,g" /tmp/runtime.properties
+  fi
+  if [[ ! -z "${EMAIL_REPLY_TO}" ]]; then
+    echo "Templating runtime.properties email.replyTo = $EMAIL_REPLY_TO"
+    sed -i "s,  # email.replyTo = vitroAdmin@mydomain.edu,email.replyTo = $EMAIL_REPLY_TO,g" /tmp/runtime.properties
   fi
-fi
 
-# copy applicationSetup.n3 if it does not already exist in target home directory
-if [ -f /usr/local/vivo/home/config/example.applicationSetup.n3 ]; then
-  if [ ! -f /usr/local/vivo/home/config/applicationSetup.n3 ]
-  then
-    echo "Copying example.applicationSetup.n3 to /usr/local/vivo/home/config/applicationSetup.n3"
-    cp /usr/local/vivo/home/config/example.applicationSetup.n3 /usr/local/vivo/home/config/applicationSetup.n3
-  else
-    echo "Using existing /usr/local/vivo/home/config/applicationSetup.n3"
+  if [[ ! -z "${LANGUAGE_FILTER_ENABLED}" ]]; then
+    echo "Templating runtime.properties RDFService.languageFilter = $LANGUAGE_FILTER_ENABLED"
+    sed -i "s,# RDFService.languageFilter = false,RDFService.languageFilter = $LANGUAGE_FILTER_ENABLED,g" /tmp/runtime.properties
+  fi
+  if [[ ! -z "${FORCE_LOCALE}" ]]; then
+    echo "Templating runtime.properties languages.forceLocale = $FORCE_LOCALE"
+    sed -i "s,# languages.forceLocale = en_US,languages.forceLocale = $FORCE_LOCALE,g" /tmp/runtime.properties
+  fi
+  if [[ ! -z "${SELECTABLE_LOCALES}" ]]; then
+    echo "Templating runtime.properties languages.selectableLocales = $SELECTABLE_LOCALES"
+    sed -i "s/# languages.selectableLocales = en_US, es_GO/languages.selectableLocales = $SELECTABLE_LOCALES/g" /tmp/runtime.properties
   fi
+
+  echo "Copying /tmp/runtime.properties to /$VIVO_HOME_DIR/config/runtime.properties"
+  cp -r /tmp/runtime.properties $VIVO_HOME_DIR/config/runtime.properties
+else
+  echo "Using existing $VIVO_HOME_DIR/config/runtime.properties"
 fi
 
+# only move applicationSetup.n3 if it does not already exist in target home directory or reconfigure env variable true
+if [ ! -f $VIVO_HOME_DIR/config/applicationSetup.n3 ] || [[ "$RECONFIGURE" = "true" ]]
+then
+  echo "Copying /tmp/applicationSetup.n3 to $VIVO_HOME_DIR/config/applicationSetup.n3"
+  cp -r /tmp/applicationSetup.n3 $VIVO_HOME_DIR/config/applicationSetup.n3
+else
+  echo "Using existing $VIVO_HOME_DIR/config/applicationSetup.n3"
+fi
+
+cp -r /tmp/vivo.war /usr/local/tomcat/webapps/$TOMCAT_CONTEXT_PATH.war
+
+export JAVA_OPTS="${JAVA_OPTS} -Dvivo-dir=$VIVO_HOME_DIR -Droot-user-address=$ROOT_USER_ADDRESS -Ddefault-namespace=$DEFAULT_NAMESPACE -Dtdb:fileMode=$TDB_FILE_MODE"
+
+echo "Giving time for Solr to startup..."
+sleep 15
+echo "Starting Tomcat"
```

Comment:
someone recently reported this issue, thanks for this. Maybe 15 should also be a parameter/variable with 15 as default value.

Reply:
This technically shouldn't be here. It is a hack and only working around a race condition when deploying using docker-compose. I will update to provide the timeout in the docker-compose file entrypoint opposed to it being in the start script of the container which will always sleep before startup.